force_bdss/mco/optimizer_engines/weighted_optimizer_engine.py

```diff
@@ -123,9 +146,29 @@ def _weighted_optimize(self, weights):
             + "Bounds: {}".format(self.parameter_bounds)
         )
 
-        weighted_score_func = partial(self.weighted_score, weights=weights)
+        weighted_score_func = partial(
+            self.weighted_score, weights=weights)
 
-        return self._scipy_optimize(weighted_score_func)
+        # use default .optimize_function
+        optimal_point = self.optimizer.optimize_function(
+            weighted_score_func,
+            self.initial_parameter_value,
+            self.parameter_bounds)
+        optimal_kpis = self._score(optimal_point)
+
+        log.info(
+            "Optimal point : {}".format(optimal_point)
+            + "KPIs at optimal point : {}".format(optimal_kpis)
+        )
+
+        return optimal_point, optimal_kpis
```

Comment:
This code could be rewritten if `optimize_function` is implemented as a generator yielding `optimal_point`:

```python
    for optimal_point in self.optimizer.optimize_function(
            weighted_score_func,
            self.initial_parameter_value,
            self.parameter_bounds)

        optimal_kpis = self._score(optimal_point)

        log.info(
            "Optimal point : {}".format(optimal_point)
            + "KPIs at optimal point : {}".format(optimal_kpis)
        )

        yield optimal_point, optimal_kpis
```